### PR TITLE
[GLUTEN-7600][VL] Prepare test case for the removal of workaround code for empty schema batches

### PR DESCRIPTION
**Pull request by zhztheplayer**
Part of #7600 

Add a unit test to guard against the removal of special handling code in query planner for empty schema batches. E.g., `EmptySchemaWorkaround` and `OffloadHashAggregate`.

Also with some code cleanups against some other unrelated PRs recently merged to main.

### CODE REVIEW

File: backends-velox/src/test/scala/org/apache/gluten/execution/MiscOperatorSuite.scala [hunk 1]
```diff
@@ -922,6 +922,18 @@ class MiscOperatorSuite extends VeloxWholeStageTransformerSuite with AdaptiveSpa
...
   }
 
+  test("count(1) on csv scan") {
+    val df = runAndCompare("select count(1) from student") {
+      val filePath = rootPath + "/datasource/csv/student.csv"
+      val df = spark.read
+        .format("csv")
+        .option("header", "true")
+        .load(filePath)
+      df.createOrReplaceTempView("student")
```
Comment (zhztheplayer) [inline backends-velox/src/test/scala/org/apache/gluten/execution/MiscOperatorSuite.scala:932]: nit: The table creation code is duplicated in different cases. Refactor required.

**Review by JkSelf — APPROVED** [review]

**Comment by zhztheplayer** [comment]
cc @zzcclp @baibaichen CH CI failure is not related
